packages/react/src/auto/polaris/inputs/PolarisBelongsToInput.tsx

```diff
@@ -0,0 +1,56 @@
+import { AnyModelManager, assert } from "@gadgetinc/api-client-core";
+import { Select } from "@shopify/polaris";
+import React from "react";
+import { useController } from "react-hook-form";
+import { GadgetBelongsToConfig } from "src/internal/gql/graphql.js";
+import { useApi } from "../../../GadgetProvider.js";
+import { useFindMany } from "../../../useFindMany.js";
+import { useFormFields } from "../../AutoForm.js";
+import { useAutoFormMetadata } from "../../AutoFormContext.js";
+
+export const PolarisBelongsToInput = (props: { field: string }) => {
+  const { metadata } = useAutoFormMetadata();
+  const api = useApi();
+  const fields = useFormFields(metadata, {});
+
+  const fieldMetadata = fields.find((field) => field[1].apiIdentifier === props.field);
+
+  if (!fieldMetadata) {
+    throw new Error(`Field ${props.field} not found in metadata`);
+  }
+
+  const path = fieldMetadata[0];
+  const _field = fieldMetadata[1];
+
+  const {
+    field: fieldProps,
+    fieldState: { error: fieldError },
+  } = useController({
+    name: path + ".id",
+    rules: { required: _field.requiredArgumentForInput },
+  });
+
+  const { ref: _ref, ...field } = fieldProps;
+  const config = _field.configuration as GadgetBelongsToConfig;
+
+  if (!config || !config.relatedModel) {
+    throw new Error(`Field ${props.field} not found in metadata`);
+  }
+
+  const modelManager = assert(
+    (api as any)[config.relatedModel.apiIdentifier] as AnyModelManager,
+    "no model manager found for action function"
+  );
+
+  const [{ data, fetching, error }, _refetch] = useFindMany(modelManager as any, { first: 25 });
+
+  if (fetching || !data) {
+    return <p>Loading...</p>;
```

Comment:
Doesnt have to be this PR but can we do a spinny on the select instead of a layout shift from `<p/>` => `<Select/>`

Reply:
100%